pkg/constants/attributes.go

```diff
@@ -17,10 +17,39 @@ package constants
 
 // Constants that are used in attributes on DevWorkspace elements (components, endpoints, etc.)
 const (
+	// DevWorkspaceStorageTypeAttribute defines the strategy used for provisioning storage for the workspace.
+	// If empty, the common PVC strategy is used.
+	// Supported options:
+	// - "common": Create one PVC per namespace, and store data for all workspaces in that namespace in that PVC
+	// - "async" : Create one PVC per namespace, and create a remote server that syncs data from workspaces to the PVC.
+	//             All volumeMounts used for devworkspaces are emptyDir
+	DevWorkspaceStorageTypeAttribute = "controller.devfile.io/storage-type"
+
+	// WorkspaceEnvAttribute is an attribute that specifies a set of environment variables provided by a component
+	// that should be added to all workspace containers. The structure of the attribute value should be a list of
+	// Devfile 2.0 EnvVar, e.g.
+	//
+	//   attributes:
+	//     workspaceEnv:
+	//       - name: ENV_1
+	//         value: VAL_1
+	//       - name: ENV_2
+	//         value: VAL_2
+	WorkspaceEnvAttribute = "workspaceEnv"
+
+	// ProjectCloneAttribute configures how the DevWorkspace will treat project cloning. By default, an init container
+	// will be added to the workspace deployment to clone projects to the workspace before it starts. This attribute
+	// must be applied to top-level attributes field in the DevWorkspace.
+	// Supported options:
+	// - "disable" - Disable automatic project cloning. No init container will be added to the workspace and projects
+	//               will not be cloned into the workspace on start.
+	ProjectCloneAttribute = "controller.devfile.io/project-clone"
```

Comment:
This is optional but I wonder if it makes more sense to do:
```
"controller.devfile.io/clone-project"
```
that way its naturally a boolean and you don't need to specify if its disabled or not. You can just specify whether its true or false (or doesn't exist at all)

Reply:
The idea with this naming was to leave it open in case there's a need for using a custom image here -- i.e. this version leaves it open to potentially supporting something like
```yaml
attributes:
    controller.devfile.io/project-clone: quay.io/amisevsk/my-custom-project-clone:latest
```
but now that I think of it, maybe it just makes sense to keep it simple and use a separate label in the off chance we need to support this. I'll update.